File: dotnet/DocumentIntelligence.Api/Storage/AzureBlobStorage.cs
namespace DocumentIntelligence.Api.Storage;

/// <summary>Azure Blob Storage implementation — stub, not yet implemented.</summary>
public class AzureBlobStorage : IFileStorage
{
    public Task<string> SaveAsync(string fileId, string filename, byte[] fileBytes, CancellationToken ct = default)
        => throw new NotImplementedException("AzureBlobStorage not implemented");

    public Task<(byte[] Bytes, string Filename)?> GetAsync(string fileId, CancellationToken ct = default)
        => throw new NotImplementedException("AzureBlobStorage not implemented");

    public Task<bool> DeleteAsync(string fileId, CancellationToken ct = default)
        => throw new NotImplementedException("AzureBlobStorage not implemented");
}

File: app/storage/azure_blob_storage.py
from app.storage.base import FileStorage


class AzureBlobStorage(FileStorage):
    """Azure Blob Storage implementation — stub, not yet implemented."""

    async def save(self, file_id: str, filename: str, file_bytes: bytes) -> str:
        raise NotImplementedError("AzureBlobStorage not implemented")

    async def get(self, file_id: str) -> bytes | None:
        raise NotImplementedError("AzureBlobStorage not implemented")

    async def delete(self, file_id: str) -> bool:
        raise NotImplementedError("AzureBlobStorage not implemented")

    def get_filename(self, file_id: str) -> str | None:
        raise NotImplementedError("AzureBlobStorage not implemented")
